src/types/list.rs

```diff
@@ -279,12 +278,12 @@ impl<'py> PyListMethods<'py> for Bound<'py, PyList> {
     /// });
     /// ```
     fn get_item(&self, index: usize) -> PyResult<Bound<'py, PyAny>> {
-        unsafe {
-            // PyList_GetItem return borrowed ptr; must make owned for safety (see #890).
-            ffi::PyList_GetItem(self.as_ptr(), index as Py_ssize_t)
-                .assume_borrowed_or_err(self.py())
-                .map(Borrowed::to_owned)
+        let py = self.py();
+        let result = unsafe { ffi::compat::PyList_GetItemRef(self.as_ptr(), index as Py_ssize_t) };
+        if !result.is_null() {
+            return Ok(unsafe { result.assume_owned(py) });
         }
+        Err(PyErr::fetch(py))
```

Comment:
We can use `.assume_owned_or_err()` to push the null-checking down:

```suggestion
        unsafe {
            ffi::compat::PyList_GetItemRef(self.as_ptr(), index as Py_ssize_t)
                .assume_owned_or_err(self.py())
        }
```